crates/driver/src/infra/api/mod.rs

```diff
@@ -69,6 +70,16 @@ impl Api {
             let router = routes::solve(router);
             let router = routes::reveal(router);
             let router = routes::settle(router);
+
+            let mut bad_tokens = bad_tokens::Detector::default()
+                .with_config(solver.bad_token_detection().tokens_supported.clone());
```

Comment:
Wouldn't it be a bit more straightforward to define this logic in a constructor, so building an instance from the config only, and all the defaults will be allocated inside that constructor?

Reply:
I think what complicates this is that the constructor would have to be aware that all these instances should share the `simulation::Detector`. All the options that come to mind seem like more hassle than they are worth:
* passing it in (same thing we have to do now)
* global singleton for the shared detector
* factory which keeps an instance of the shared detector internally

I remove `.with_config()` and put it directly into the constructor if you like, though.

Reply:
My suggestion was only about the `with_config()` function. `with_simulation_detector()` remains unchanged.